api/src/main/java/org/motechproject/nms/api/web/UserController.java

```diff
@@ -256,13 +257,12 @@ private UserResponse getFrontLineWorkerResponseUser(String serviceName, Long cal
             }
         }
 
-        if (flw == null) {
-            // New requirement - https://applab.atlassian.net/projects/NMS/issues/NMS-325 - Block anonymous FLWs
-            // if flw is null here, we don't already have a record from MCTS. return 403
-            if (MOBILE_ACADEMY.equals(serviceName)) {
-                throw new NotAuthorizedException(String.format(NOT_AUTHORIZED, CALLING_NUMBER));
-            }
-        } else {
+        if (MOBILE_ACADEMY.equals(serviceName)) {
+            // make sure that flw is authorized to use MA
+            restrictAnonymousMAUserCheck(flw);
```

Comment:
@frankhuster Which part? I don't think we can assume that the service is MA here -- could be MK (although if MK ever comes online we'll probably be asked to disable anonymous access there too...)

Or were you referring to the restrictAnonymousUserCheck? There's a case where the flw might be non-null but the user is ANONYMOUS (because we have created FLW records for non-MCTS FLWs who have started the course prior to these changes).